### PR TITLE
raftstore: convert socket timeout error.

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @BusyJay @tiancaiamao

### CODE REVIEW

**Comment by tiancaiamao** [comment]
LGTM but this may be a temporary solution. 

"Resource temporarily unavailable" error message has more complex reason, can't get socket data to read is just one of the possibility.
I'm not sure interrupted by signal or other kinds, but make the code runs on all "Unix-like" system, this change is  far from complete.

File: tests/raftstore/server.rs [hunk 1]
```diff
@@ -167,7 +167,21 @@ impl Simulator for ServerCluster {
 
         conn.set_read_timeout(Some(timeout)).unwrap();
         let mut resp_msg = Message::new();
-        let get_msg_id = try!(rpc::decode_msg(&mut conn, &mut resp_msg));
+        let get_msg_id = try!(rpc::decode_msg(&mut conn, &mut resp_msg).map_err(|e| {
+            if let CodecError::Io(ref err) = e {
+                // If read timeout, the error will throw "Resource temporarily unavailable",
```
Comment (BusyJay) [inline tests/raftstore/server.rs]: I think we can check if it's timeout by checking `elapsed > timeout` rather than just message content or error code.

**Comment by ngaut** [comment]
It's not a correct solution. We should fix it cleanly.

File: tests/raftstore/server.rs (hunk 1 above)
Comment (siddontang) [inline tests/raftstore/server.rs]: For socket, this message is certainly corresponding to EAGAIN and EWOULDBLOCK for different  system, so we can use error code and message to check this only.

**Comment by siddontang** [comment]
No, we should check timeout and covert to Timeout error for outer use, and the timeout for this case returns this message.

File: tests/raftstore/server.rs (hunk 1 above)
Comment (BusyJay) [inline tests/raftstore/server.rs]: > EAGAIN or EWOULDBLOCK
The socket is marked nonblocking and the receive operation would block, or a receive timeout had been set and the timeout expired before data was received.  POSIX.1 allows either error to be returned for this case, and does not require these constants to have the same value, so a portable application should check for both possibilities.

Quoted from man `recv`.

**Comment by siddontang** [comment]
PTAL

**Comment by ngaut** [comment]
LGTM

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**